inst/include/Rcpp/platform/compiler.h

```diff
@@ -57,10 +44,11 @@
         #define IS_GCC_460_OR_LATER
     #endif
     // Inform users of dated compiler (e.g. <= RHEL6 )
-    #if GCC_VERSION <= 40602
-        #pragma message ("WARNING: GCC Compiler Version <= 4.6.2")
-        #pragma message ("WARNING: The compiler being used is both OUTDATED and INCOMPLETE in terms of C++ standards;")
-        #pragma message ("WARNING: if something does not work, you should upgrade your compiler!")
+    #if GCC_VERSION <= 40602 && !defined(RCPP_GCC_COMPILER_WARNING_SILENT)
+         // Ensure not fake GCC flag set by clang or intel.
+         #if !(defined(__clang__) || defined(__INTEL_COMPILER))
```

Comment:
Are we saying that anyone running RHEL 6 is going to see this warning when
compiling Rcpp? That I think is far too aggressive as this describes the
vast majority of Linux systems deployed in larger organizations. Worse yet
is that the end users who see this message during compilation have no way
to update the compiler (that's handled by the sysadmin on a multi-user
system, and compiler upgrades are considered extremely heavyweight/risky so
are basically not done).



On Sat, Jul 30, 2016 at 10:13 PM, James J Balamuta <notifications@github.com
> wrote:

> In inst/include/Rcpp/platform/compiler.h
> <https://github.com/RcppCore/Rcpp/pull/515#discussion_r72896929>:
>
> > @@ -57,10 +44,11 @@
> >          #define IS_GCC_460_OR_LATER
> >      #endif
> >      // Inform users of dated compiler (e.g. <= RHEL6 )
> > -    #if GCC_VERSION <= 40602
> > -        #pragma message ("WARNING: GCC Compiler Version <= 4.6.2")
> > -        #pragma message ("WARNING: The compiler being used is both OUTDATED and INCOMPLETE in terms of C++ standards;")
> > -        #pragma message ("WARNING: if something does not work, you should upgrade your compiler!")
> > +    #if GCC_VERSION <= 40602 && !defined(RCPP_GCC_COMPILER_WARNING_SILENT)
> > +         // Ensure not fake GCC flag set by clang or intel.
> > +         #if !(defined(__clang__) || defined(__INTEL_COMPILER))
>
> The reason why I added in the detect genuine GCC is due to the warning
> being triggered on macOS due to GCC headers being identified as 4.2.1
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/RcppCore/Rcpp/pull/515/files/c5f7b40770c14f8efb12f2f5e77e7aa5d17e0d0d..6b292afd4f11c5f9f95abd830f5cb383a194c3de#r72896929>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAGXx8hmEGS19KROXcv6n0iq4lmyLblbks5qbASvgaJpZM4JTRmr>
> .
>